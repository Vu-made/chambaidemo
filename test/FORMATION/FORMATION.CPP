#include <bits/stdc++.h>
#define N 1000001
using namespace std;
int L[N], R[N];
int n, m;
void xuly1()
{
    int i,j;
    cin >> i >> j;
    L[i] = L[j];
    R[i] = j;
    R[L[j]] = i;
    L[j] = i;
}

void xuly2()
{
    int i,j;
    cin >> i >> j;
    L[i] = j;
    R[i] = R[j];
    L[R[j]] = i;
    R[j] = i;
}

void xuly3()
{
    int i;
    cin >> i;
    R[L[i]] = R[i];
    L[R[i]] = L[i];
}

void xuly4()
{
    int i;
    cin >> i;
    cout << L[i] << ' ' << R[i] << endl;
}
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    freopen("FORMATION.INP","r",stdin);
    freopen("FORMATION.OUT","w",stdout);
    cin >> n >> m;
    for (int i=1; i<=m; i++)
    {
        int k;
        cin >> k;
        if (k == 1) xuly1();
        else if (k==2) xuly2();
        else if (k==3) xuly3();
        else xuly4();
    }
    return 0;
}
