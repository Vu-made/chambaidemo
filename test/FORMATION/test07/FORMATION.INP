47 72
2 30 1
3 30
1 35 1
4 1
3 1
4 35
2 19 35
3 35
4 19
1 12 19
1 4 19
2 5 19
3 4
2 17 19
3 5
2 32 19
1 3 17
2 10 17
2 2 19
3 10
1 23 12
1 18 32
1 43 2
1 11 3
2 26 23
3 26
1 20 2
4 20
2 21 12
3 32
1 42 23
2 44 18
4 17
1 30 12
2 39 44
3 43
3 11
2 5 30
2 26 18
1 27 21
1 7 27
1 32 39
1 16 2
2 9 42
3 20
4 12
3 17
1 22 16
4 44
3 42
4 39
4 9
3 12
3 9
3 16
2 40 2
1 25 44
3 27
3 39
2 14 40
3 40
3 7
1 27 14
3 2
3 14
4 3
3 3
3 22
3 44
3 30
1 45 23
1 14 19
