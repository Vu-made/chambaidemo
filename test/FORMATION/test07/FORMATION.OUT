35 0
0 0
0 0
43 2
3 0
5 7
26 32
32 3
0 23
32 0
